src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useIsApprovalNeeded.tsx

```diff
@@ -11,44 +12,49 @@ import { GetAllowanceErr } from './helpers'
 export const useIsApprovalNeeded = (
   tradeQuoteStep: TradeQuoteStep | undefined,
   sellAssetAccountId: AccountId | undefined,
-  watch: boolean,
 ) => {
-  const { isLoading, data } = useAllowance(tradeQuoteStep, sellAssetAccountId, watch)
-
-  const isApprovalNeeded = useMemo(() => {
-    if (tradeQuoteStep === undefined) return undefined
-
-    const isEvmChainId = evmChainIds.includes(tradeQuoteStep.sellAsset.chainId as EvmChainId)
-
-    if (!isEvmChainId) return false
-
-    if (data?.isErr()) {
-      const error = data.unwrapErr()
-      // the error type is a GetAllowanceErr enum so we can handle all cases with exhaustiveness
-      // checking to prevent returning the wrong value if we add more error cases
-      switch (error) {
-        case GetAllowanceErr.IsFeeAsset:
-        case GetAllowanceErr.NotEVMChain:
-          // approval not required
-          return false
-        case GetAllowanceErr.MissingArgs:
-          // not known yet
-          return undefined
-        default:
-          assertUnreachable(error)
+  const selectIsApprovalNeeded = useCallback(
+    (data: Result<string, GetAllowanceErr>) => {
+      if (tradeQuoteStep === undefined) return undefined
+
+      if (data?.isErr()) {
```

Comment:
we can leverage `selectAllowanceCryptoBaseUnit`

```ts
      const allowanceCryptoBaseUnit = selectAllowanceCryptoBaseUnit(data)
      if (allowanceCryptoBaseUnit === undefined) return false

      return bn(allowanceCryptoBaseUnit).lt(
        tradeQuoteStep.sellAmountIncludingProtocolFeesCryptoBaseUnit,
      )
      ```

Reply:
🧠  445cdde269d2023728b8435b25291fe409ffd883